sdk/python/feast/feature_store.py

```diff
@@ -594,8 +618,10 @@ def _entity_row_to_field_values(
     return result
 
 
-def _group_refs(
-    feature_refs: List[str], all_feature_views: List[FeatureView]
+def _validate_and_group_feature_refs(
```

Comment:
Previous discussion for reference https://github.com/feast-dev/feast/pull/1641#discussion_r655780532

Reply:
@woop thanks for the reference.

@Mwad22
> The logic in _group_refs lends itself well to checking for feature name collisions, but more importantly I think placing the check here is reasonable considering the description of the _group_refs function: """ Get list of feature views and corresponding feature names based on feature references""".
> 
> If the feature names are invalid (e.g. have a collision) this would be a fitting place to throw the error, I believe. Do let me know if you disagree.

The purpose of `_group_refs` is to group features by FeatureView. The method works fine even if there are collisions. This is more obvious if you look at the modified lines: no line is removed and no existing variable inside the function touched. The new code only introduces new variables and adds an additional logic for handling collisions. If _group_refs intrinsically wouldn't work during collisions, I'd agree with you. But for these reasons, I think it still makes sense to split the methods into validation & grouping.

Reply:
Sounds good,  I just committed the changes requested here and across the other comments. Thank you for the detailed explanations, they make a lot of sense and are appreciated on my end!